state/accountsEphemeralProvider.go

```diff
@@ -0,0 +1,92 @@
+package state
+
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+)
+
+// AccountsEphemeralProvider acts as an "ephemeral" provider for accounts.
+// Make sure such a provider isn't reused among multiple transactions selections or multiple processing (of blocks) phases,
+// since it contains a **never-invalidating cache** (deliberate, by design).
+// Create it privately (don't receive it in constructors), use it privately, make sure it's forgotten afterwards. Don't keep lasting references to it.
+// Note: this structure is exported on purpose (less ceremonious code where it's being used, no extra interfaces needed).
+type AccountsEphemeralProvider struct {
```

Comment:
Most of the logic extracted from `process/block/preprocess/selectionSession.go`.